### PR TITLE
Add model equations for simple dead end filtration model

### PR DESCRIPTION
**Pull request by daklauss**
Reopened Pull request after i merged too fast.

First implementation of the easy def model as described in https://github.com/cadet/CADET-Python-Simulator/issues/5, stil wip

TODOS:

- [x] Tests
    - [x] Residual tests
    - [x] Test unit Operation
 - [x] Considering outsourcing to rejection base class?
    - [x]  Use rejection class in def
    - [x]  Test rejection class?

### CODE REVIEW

File: CADETPythonSimulator/residual.py [hunk 1]
```diff
@@ -77,6 +77,72 @@ def calculate_residual_visc_cstr():
...
-    raise NotImplementedError
+def calculate_residual_cake_vol_def(
+        V_dot_f : float,
+        rejection : np.ndarray,
+        molar_volume : np.ndarray,
+        c_in : np.ndarray,
+        V_dot_C : float
+        ) -> float:
+    """
+    Residual equation for the Volume
```
Comment (schmoelder) [inline CADETPythonSimulator/residual.py:88]: [Clarification] For which volume is this equation? The cake or the permeate?

File: CADETPythonSimulator/residual.py [hunk 2]
```diff
@@ -77,6 +77,72 @@ def calculate_residual_visc_cstr():
...
+    V_dot_f : float
+        Flowrate of incoming feed
+    rejection : float
+        Rejection of the filter
+    gamma : float
+        Portion of suspended material
+    V_dot_C : float
+        Change of Cake Volume
+    """
+
```
Comment (schmoelder) [inline CADETPythonSimulator/residual.py:101]: [Formatting] Remove empty line.

**Review by schmoelder — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -68,4 +68,8 @@ version = { attr = "CADETPythonSimulator.__version__" }
 [tool.ruff]
 # Same as Black.
 line-length = 88
-indent-width = 4
```
[the remark this answers is not in the document]
Comment (daklauss) [reply pyproject.toml:71]: [Question] Done, is there a way for ruff to check it in the pyproject.toml. For regular .py files i get warnings.

File: CADETPythonSimulator/residual.py (hunk 1 above)
Comment (daklauss) [reply CADETPythonSimulator/residual.py]: Cake

File: pyproject.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (schmoelder) [reply pyproject.toml:71]: This should be covered by [W292](https://docs.astral.sh/ruff/rules/missing-newline-at-end-of-file/).

**Pull request merged**